### ISSUE TITLE
Nomenclature for build scripts?

### ISSUE THREAD

**Issue by aceakash** [mergermarket/goMakeIt#1]
Wherever possible, it will be nice to provide scripts to do the same thing with or without Docker.

e.g
```
# run unit tests with locally installed Go
./run-unit-tests.sh

# same as above, but in Docker
./docker-run-unit-tests.sh

# extend to other scripts
./run-against-ci.sh
./docker-run-against-ci.sh
```

Thoughts?

**Comment by quii** [comment]
My worry with multiple ways of building the software is that it's two things to maintain and as things change in my experience over time things stop working for one of the methods.

Docker is how we will build in Jenkins so in my view we should optimise to that. But I appreciate there can be pain with docker, but we should just fix that. When you build with docker once the image is made the build is basically as fast as normal.

**Comment by quii** [comment]
Just to add, here are the build times. `test.sh` runs `build-app` inside a container

````
MML242:goMakeIt chrisj$ time ./build-app.sh 
=== RUN   TestHelloRouter
--- PASS: TestHelloRouter (0.00s)
PASS
coverage: 57.1% of statements
ok  	github.com/mergermarket/goMakeIt	0.010s

real	0m0.753s
user	0m0.598s
sys	0m0.253s
MML242:goMakeIt chrisj$ time ./test.sh 
=== RUN   TestHelloRouter
--- PASS: TestHelloRouter (0.00s)
PASS
coverage: 57.1% of statements
ok  	github.com/mergermarket/app	0.007s

real	0m1.992s
user	0m0.271s
sys	0m0.077s
````

**Comment by aceakash** [comment]
I'm not so concerned about build times - it's more about being on a poor network and having to wait ages to download the Docker images.

I meant something like this: https://github.com/mergermarket/authentication-service/blob/master/docker-run-unit-tests-with-coverage.sh

The `docker-do-x.sh` script just builds the docker image and calls `./do-x.sh` --- so changes only happen in `do-x.sh`.

**Comment by quii** [comment]
Yeah, perhaps you could just rename what we have if that would make it clearer. The docker image just runs `build-app.sh` :)
